x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -1659,6 +1659,68 @@ export class RulesClient {
     );
   }
 
+  public async unsnooze({ id }: { id: string }): Promise<void> {
+    return await retryIfConflicts(
+      this.logger,
+      `rulesClient.unsnooze('${id}')`,
+      async () => await this.unsnoozeWithOCC({ id })
+    );
+  }
+
+  private async unsnoozeWithOCC({ id }: { id: string }) {
+    const { attributes, version } = await this.unsecuredSavedObjectsClient.get<RawRule>(
+      'alert',
+      id
+    );
+
+    try {
+      await this.authorization.ensureAuthorized({
+        ruleTypeId: attributes.alertTypeId,
+        consumer: attributes.consumer,
+        operation: WriteOperations.Unsnooze,
+        entity: AlertingAuthorizationEntity.Rule,
+      });
```

Comment:
I haven't tested this, but it appears that the authorization check will never occur if the user tries to unsnooze an alert, and the alert is not found, then it will throw a 404 error before the authZ check occurs (and thus, it will never get audited).

It looks like the rest of the rules client functions the same way, too. Can you confirm? CC @XavierM